code/modules/mining/equipment/wormhole_jaunter.dm

```diff
@@ -36,47 +37,63 @@
 
 	return destinations
 
-/obj/item/wormhole_jaunter/proc/activate(mob/user, adjacent)
+/obj/item/wormhole_jaunter/proc/can_jaunter_teleport()
+	var/list/destinations = get_destinations()
+	return destinations.len > 0
+
+/obj/item/wormhole_jaunter/proc/activate(mob/user, adjacent, teleport)
 	if(!turf_check(user))
 		return
 
-	var/list/L = get_destinations(user)
-	if(!L.len)
-		to_chat(user, span_notice("The [src.name] found no beacons in the world to anchor a wormhole to."))
-		return
-	var/chosen_beacon = pick(L)
-	var/obj/effect/portal/jaunt_tunnel/J = new (get_turf(src), 100, null, FALSE, get_turf(chosen_beacon))
-	if(adjacent)
-		try_move_adjacent(J)
+	if(!can_jaunter_teleport())
+		if(user)
+			to_chat(user, span_notice("\The [src] found no beacons in the world to anchor a wormhole to."))
+		else
+			visible_message(span_notice("\The [src] found no beacons in the world to anchor a wormhole to!"))
+		return TRUE
+
+	var/list/destinations = get_destinations()
+	var/chosen_beacon = pick(destinations)
+
+	var/obj/effect/portal/jaunt_tunnel/tunnel = new (get_turf(src), 100, null, FALSE, get_turf(chosen_beacon))
+	if(teleport)
+		tunnel.teleport(user)
+	else if(adjacent)
+		try_move_adjacent(tunnel)
+
 	playsound(src,'sound/effects/sparks4.ogg',50,TRUE)
 	qdel(src)
+	return FALSE
 
 /obj/item/wormhole_jaunter/emp_act(power)
 	. = ..()
 	if(. & EMP_PROTECT_SELF)
 		return
 
+	var/triggered = FALSE
+	if(power == 1)
+		triggered = TRUE
+	else if(power == 2 && prob(50))
+		triggered = TRUE
+
 	var/mob/M = loc
-	if(istype(M))
-		var/triggered = FALSE
-		if(M.get_item_by_slot(ITEM_SLOT_BELT) == src)
-			if(power == 1)
-				triggered = TRUE
-			else if(power == 2 && prob(50))
-				triggered = TRUE
-
-		if(triggered)
-			M.visible_message(span_warning("[src] overloads and activates!"))
-			SSblackbox.record_feedback("tally", "jaunter", 1, "EMP") // EMP accidental activation
-			activate(M)
+	if(istype(M) && triggered)
+		M.visible_message(span_warning("Your [src] overloads and activates!"))
+		SSblackbox.record_feedback("tally", "jaunter", 1, "EMP") // EMP accidental activation
+		activate(M, FALSE, TRUE)
+	else if(triggered)
+		visible_message(span_warning("\The [src] overloads and activates!"))
+		activate()
 
 /obj/item/wormhole_jaunter/proc/chasm_react(mob/user)
-	if(user.get_item_by_slot(ITEM_SLOT_BELT) == src)
-		to_chat(user, span_notice("Your [name] activates, saving you from the chasm!"))
+	var/can_jaunter_teleport = activate(user, FALSE, TRUE)
+	if(can_jaunter_teleport)
+		to_chat(user, span_notice("Your [src] activates, saving you from the chasm!"))
+		user.visible_message(span_boldwarning("[user] falls into the chasm!")) //To freak out any bystanders
```

Comment:
I think this is meant to display on the person falling. it should prob display on the turf they were on pre activate, since I'm pretty sure at this point in the call chain the user has already moved to their beacon

Reply:
This should resolve the above but I won't be able to test it myself until tomorrow. 
 https://github.com/tgstation/tgstation/pull/61055/commits/155bd375bd769d37b53ee4a4fb5fad46dcb69974

Reply:
You'll want to send it on the turf the player was on before they fell. so get the turf with get_turf(user) before the fall, and then use it as the thing to send the message on when they teleport

Reply:
Look at the chasm changes in the commit.  It broadcasts the message to nearby viewers instead of doing it in the jaunter.

Reply:
you're gonna need to call visible_message on a thing that's in the world, since it's defined on /atom and components are datums

Reply:
Thanks for the info.  Implemented with https://github.com/tgstation/tgstation/pull/61055/commits/ab1e566f0abf5d6b0bb7fbe22f9b052f3b49aa60